morpho5/geometry/functional.c

```diff
@@ -1530,8 +1525,8 @@ value Hydrogel_gradient(vm *v, int nargs, value *args) {
     value out=MORPHO_NIL;
     hydrogelref ref;
     if (functional_validateargs(v, nargs, args, &info)) {
-        if (hydrogel_prepareref(MORPHO_GETINSTANCE(MORPHO_SELF(args)), info.mesh, MESH_GRADE_VOLUME, info.sel, &ref)) { 
-            info.g = MESH_GRADE_VOLUME;
+        if (hydrogel_prepareref(MORPHO_GETINSTANCE(MORPHO_SELF(args)), info.mesh, -1, info.sel, &ref)) {
```

Comment:
I am passing the grade value to `hydrogel_prepareref` as `-1` so that the value stored in ref->grade will be the mesh's maxgrade. I don't know if hard-coding this `-1` is the best way to go though.